File: DWGetDatasetInfo/src/org/knime/data/world/restful/GetDatasetInfoNodeFactory.java
package org.knime.data.world.restful;

import org.knime.core.node.NodeDialogPane;
import org.knime.core.node.NodeFactory;
import org.knime.core.node.NodeView;

/**
 * <code>NodeFactory</code> for the "GetDatasetInfo" Node.
 * Gets info on a data.world dataset via the data.world resful api.
 *
 * @author Appliomics, LLC
 */
public class GetDatasetInfoNodeFactory 
        extends NodeFactory<GetDatasetInfoNodeModel> {

    /**
     * {@inheritDoc}
     */
    @Override
    public GetDatasetInfoNodeModel createNodeModel() {
        return new GetDatasetInfoNodeModel();
    }

    /**
     * {@inheritDoc}
     */
    @Override
    public int getNrNodeViews() {
        return 1;
    }

    /**
     * {@inheritDoc}
     */
    @Override
    public NodeView<GetDatasetInfoNodeModel> createNodeView(final int viewIndex,
            final GetDatasetInfoNodeModel nodeModel) {
        return new GetDatasetInfoNodeView(nodeModel);
    }

    /**
     * {@inheritDoc}
     */
    @Override
    public boolean hasDialog() {
        return true;
    }

    /**
     * {@inheritDoc}
     */
    @Override
    public NodeDialogPane createNodeDialogPane() {
        return new GetDatasetInfoNodeDialog();
    }

}

File: DWGetDatasetInfo/src/org/knime/data/world/restful/GetDatasetInfoNodeFactory.scala
package org.knime.data.world.restful

import org.knime.core.node.NodeDialogPane
import org.knime.core.node.NodeFactory
import org.knime.core.node.NodeView

/**
 * <code>NodeFactory</code> for the "GetDatasetInfo" Node.
 * 
 *
 * @author Appliomics, LLC
 */
class GetDatasetInfoNodeFactory extends NodeFactory[GetDatasetInfoNodeModel] {
  
  /**
   * {@inheritDoc}
   */
  override def createNodeModel() : GetDatasetInfoNodeModel = new GetDatasetInfoNodeModel

  /**
   * {@inheritDoc}
   */
  override def getNrNodeViews() : Int = 0

  /**
   * {@inheritDoc}
   */
  override def createNodeView(viewIndex : Int, nodeModel : GetDatasetInfoNodeModel)
    : NodeView[GetDatasetInfoNodeModel] = new GetDatasetInfoNodeView(nodeModel)

  /**
   * {@inheritDoc}
   */
  override def hasDialog() : Boolean = true

  /**
   * {@inheritDoc}
   */
  override def createNodeDialogPane() : NodeDialogPane = new GetDatasetInfoNodeDialog

}

